tlsutil/config.go

```diff
@@ -351,7 +351,7 @@ func (c *Config) baseVerifyIncoming() bool {
 
 func loadKeyPair(certFile, keyFile string) (*tls.Certificate, error) {
 	if certFile == "" || keyFile == "" {
-		return &tls.Certificate{}, nil
+		return nil, nil
```

Comment:
Returning an empty cert here led to all kinds of strange things. This is gone now for good.